crates/assistant_tooling/src/tool.rs

```diff
@@ -149,12 +118,10 @@ pub trait LanguageModelTool {
     /// Execute the tool
     fn execute(&self, input: &Self::Input, cx: &AppContext) -> Task<Result<Self::Output>>;
```

Comment:
Since we're going this route with a View, it's tempting to make `execute` return a `Task<Result<Self::View>>`.